src/backend/api/meals.js

```diff
@@ -1,15 +1,97 @@
+const { response, request } = require("express");
 const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
+
+    
+    if('maxPrice' in request.query){
+      const maxPrice = request.query.maxPrice;
+      const highPriceMeals = await knex ('meal').where ('price', '<=', maxPrice);
+      response.send(highPriceMeals);
+      return;
+    }
+    if('availableReservations' in request.query){
+      const availableReservations = (request.query.availableReservations == 'true');
+      const numberOfReservations = await knex.from('meal').join('reservations','meal.id', '=', 'reservations.id')
```

Comment:
not sure it's correct. i think the database should be `meals` and not `meal` if you generated the database from the `database_dump_meals.sql`
```suggestion
      const numberOfReservations = await knex.from('meal').join('reservations','id', '=', 'reservations.id')
```